pkg/modprovider/tfmodules.go

```diff
@@ -220,6 +220,11 @@ func latestModuleVersion(ctx context.Context, moduleSource string) (*version.Ver
 	switch parsed := parsedSource.(type) {
 	case addrs.ModuleSourceRegistry:
 		source = parsed
+	case addrs.ModuleSourceRemote:
+		// All other valid remote module sources do not support a separate version field
+		// Opentofu will resolve the source by remote address alone.
+		// See https://opentofu.org/docs/language/modules/sources
+		return &version.Version{}, nil
```

Comment:
This is a little weird to have a pointer to an empty Version but it seems to work out given how this is used in https://github.com/pulumi/pulumi-terraform-module-provider/blob/6478a47b41507867f9dfbea7f26c96c4bf75d83c/pkg/modprovider/server.go#L155